### PR TITLE
Update `PR Template`

### LINKED ISSUE #1242: Update the checklist in the PR template
(Will add more details later)

### PR DESCRIPTION
**Pull request by playfulFence**
closes #1242 

Removing parts, which are now checked in `CI`
Add something to increase reviewers QoL, like `Testing` section and short `Description`

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-### Must
 
-- [ ] The code compiles without `errors` or `warnings`.
-- [ ] All examples work.
-- [ ] `cargo fmt` was run.
-- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
-- [ ] You updated existing examples or added examples (if applicable).
-- [ ] Added examples are checked in CI
+### Submission Checklist 📝
+- [ ] Examples related to your contribution work as expected
```
Comment (bjoernQ) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Maybe "Examples related to your contribution work as expected" and "I have updated existing examples or added examples (if applicable)." is a bit redundant?

**Review by SergioGasquez — COMMENTED** [review]
A few comments, feel free to disregard them, they are just my 2 cents

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
-### Nice to have
+### Pull Request Details 📖
 
-- [ ] You add a description of your work to this PR.
-- [ ] You added proper docs for your newly added features and code.
+#### Testing
+Describe how you tested your changes.
+
+#### Description
+Please provide a clear and concise description of your changes, including the motivation behind these changes. The context is crucial for the reviewers.
```
[the remark this answers is not in the document]
Comment (playfulFence) [reply .github/PULL_REQUEST_TEMPLATE.md:20]: Agree, thanks

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 
-- [ ] The code compiles without `errors` or `warnings`.
-- [ ] All examples work.
-- [ ] `cargo fmt` was run.
-- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
-- [ ] You updated existing examples or added examples (if applicable).
-- [ ] Added examples are checked in CI
+### Submission Checklist 📝
+- [ ] Examples related to your contribution work as expected
+- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the proper section.
```
[the remark this answers is not in the document]
Comment (playfulFence) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: The only reason it's here is the "PROPER SECTION" part, that is why that point looks legit to me, not the "CHANGELOG" part, which, as you said, is already checked in the CI.
@MabezDev suggested to add _Italics_ to make an accent on this, what do you think?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (playfulFence) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Yes, maybe we should go with only one of them. 
"I have updated existing examples or added examples (if applicable)" seems to "cover" most of wanted PR characteristic 🤔

**Review by MabezDev — APPROVED** [review]
LGTM, thank you!

**Pull request merged**